### PR TITLE
docs(AsyncPipe): API updates + new example

### PR DESCRIPTION
**Pull request by chalin**
The new example is based on TS example sources.
No tests have been written yet.

### CODE REVIEW

**Comment by chalin** [comment]
cc @kwalrath

File: lib/src/common/pipes/async_pipe.dart [hunk 1]
```diff
@@ -39,22 +39,12 @@ var _observableStrategy = new ObservableStrategy();
 Future<dynamic> ___unused;
 
 /**
- * The `async` pipe subscribes to an Observable or Promise and returns the latest value it has
- * emitted.
- * When a new value is emitted, the `async` pipe marks the component to be checked for changes.
+ * An `async` pipe awaits for a value from a [Future] or [Stream]. When a value is received, 
+ * the `async` pipe marks the component to be checked for changes.
  *
  * ### Example
```
Comment (adaojunior) [inline lib/src/common/pipes/async_pipe.dart]: The dart style guide states that you should use `///` instead of `/** ... */` in cases like this.

Comment (chalin) [inline lib/src/common/pipes/async_pipe.dart]: Right, thanks, but there are so many across the project, that I figured that we could address these in a comment-notation-clean-up pass (while making no other changes).

Comment (adaojunior) [inline lib/src/common/pipes/async_pipe.dart]: I created this PR to fix some of it.
https://github.com/dart-lang/angular2/pull/29

**Comment by kevmoo** [comment]
Please rebase on the latest `master` – we now have travis enabled.

**Comment by chalin** [comment]
FYI, I rebased both my my current PRs.

**Comment by zoechi** [comment]
It would be interesting to know what happens when two bindings are used to the same stream in one template

```dart
<div *ngFor="let item in items | async">1 - {{item}}</div>
<div *ngFor="let item in items | async">2 - {{item}}</div>
```

AFAIK Angular only subscribes once and reuses the result but it would be great to have this mentioned.

**Comment by chalin** [comment]
@zoechi : thanks, I'll give that a try and document appropriately (either here or somewhere in one of the guides).

**Comment by chalin** [comment]
Rebased.

Are you wanting examples to be under an `example` folder rather than `examples`? (I'm personally for dropping the "s", but I'm willing to go with local preferences.)

**Comment by kevmoo** [comment]
> Are you wanting examples to be under an example folder rather than examples? (I'm personally for dropping the "s", but I'm willing to go with local preferences.)

In general, yes. That's our convention.

**Comment by zoechi** [comment]
This way you should ensure that you use `--no-package-symlinks` with `pub` otherwise running `pub get` or `pub upgrade` in the main directory ruins your packages symlinks within the examples projects.

**Comment by chalin** [comment]
Thanks @zoechi!

@kevmoo - might we want to reconsider whether this API examples folder should reside in this repo or elsewhere? @ferhatb: any preference?

**Comment by chalin** [comment]
@kevmoo upgraded to beta.20. Renamed `examples` folder to `example`.

**Pull request closed without merging**